### ISSUE TITLE
Test in assistive technologies when all the MVP rebrand work is put together

### ISSUE THREAD

**Issue by hazalarpalikli** [alphagov/govuk-design-system#4616]
## What
Do our standard set of tests in various assistive technologies (JAWS, NVDA, VoiceOver on macOS and iOS, TalkBack on Android, Dragon, Windows High Contrast Mode, changing colours in Firefox).

## Why
We don’t expect any issues because accessibility has been considered in every step of our work, but we have to do our due diligence before any of the changes make it onto any public website.

## Who needs to work on this
Accessibility specialist

## Who needs to review this
Designer, developer

## Done when

- A couple of different header options have been tested and the results were captured in a spreadsheet
- If anything actionable came out of that, let a designer and/or a developer review it and decide if and how it should be fixed
- If something to be fixed came out of that, it has been fixed
- If something to maybe address later came out of that, new GitHub issues were created

**Comment by mia-allers-gds** [comment]
@selfthinker I'm wondering if you might be able to get started on this while beeps is doing the refinements — they are all small design tweaks (eg, make the font slightly bigger, move things 1px left) so if you two have a chat about whether the prototype is ready to test.

**Comment by selfthinker** [comment]
While I was testing I decided it didn't make any sense to test in our standard set of screen readers because the markup has changed only minimally. I still tested in a desktop screen reader (NVDA in Chrome) and mobile screen reader (VoiceOver on iOS) to make sure none of the CSS changes could have had an unintended consequence.
As expected, the rebrand header, footer, service nav and cookie message all behaved exactly the same as the current live versions of them.

I also tested font increase in Firefox, but that had all expected results.
I focussed more on all the **colour tests** (WHCM and Firefox settings) instead. I found the following issues with those...


## No visible distinction of header

The current header is visually clearly distinguished in force colour modes because it has a thick line underneath it. The rebranded version doesn't have any line. When there is a service navigation underneath, it's not as bad because the service nav has a line, and both could be seen as part of the same header, which is okay.
As this is a clear regression, this should be fixed. I would advise to add a transparent border (i.e. otherwise invisible except in force colour modes). What do designers think, should this border be thin or thick? (Read the next related issue below.)

Current:
<img width="900" alt="Current top of example page with thick line under header and thin line under service nav" src="https://github.com/user-attachments/assets/2be11e12-3839-4c4e-8028-c208afd8b811" />

Rebranded:
<img width="890" alt="Rebranded top of example page with no line under header and thin line under service nav" src="https://github.com/user-attachments/assets/19a5f78e-aa14-4e8e-936e-229760e64a94" />


## Inconsistent borders between cookie banner and header

The cookie banner (current as well as rebranded) has a thick transparent bottom border when in force colour mode. I assume that was added to make it look consistent with the header bottom border. Be aware that this will now look inconsistent with the rebranded header.
When adding a transparent bottom border for the issue above, consider making the cookie banner bottom border the same width for consistency. Or not, it's up to the designers in case this was not the original intention.
We don't necessarily need to fix this but as it's easy to do and is a consequence of changing the header, I think it makes sense to fix.

Current:
<img width="968" alt="Current top of example page with thick line under cookie banner and another thick line under header" src="https://github.com/user-attachments/assets/76ad4717-9b7a-4eec-a505-9845c6ec96ad" />

Rebranded:
<img width="996" alt="Rebranded top of example page with thick line under cookie banner and no line under header" src="https://github.com/user-attachments/assets/819eeda8-46e7-4af7-9d07-d8d9036b30fa" />


## Crest and crown in footer invisible with dark colours

Because the crest and crown in the footer are transparent, they become invisible when the background colour changes to something similar to their colour. This doesn't happen to the OGL logo as that has a `fill="currentColor"`.
This has always been a problem, so doesn't need to be fixed in this cycle. But because the crown is new and we have changed the crest, now we be a good time to fix that. Except the crest is a background image, which makes this particular change harder. So, maybe we should only change the crown?

Current:
<img width="1050" alt="Current footer with somewhat visible grey crest on black" src="https://github.com/user-attachments/assets/25282e67-5d75-4227-b8e6-af87140e9ebf" />

Rebranded:
<img width="997" alt="Rebranded footer with invisible black crest and crown on black" src="https://github.com/user-attachments/assets/5d1e6b6a-0bbf-41d0-9819-0e9efe6f8b79" />


## Service navigation broken in WHCM

The service navigation break into one item per line when viewed in Windows High Contrast Mode. This happens with the current as well as the rebranded version of it.

This is an already existing bug that I have discovered while testing, so nothing to fix in this cycle. I will file a proper bug report accordingly.

**Comment by selfthinker** [comment]
Because @Ciandelle is working on a kitchen sink page that includes all the different things that could be close to the header, I plan to test that page as well. Especially because the current review app version I tested didn't include any example page with the proper rebranded header and footer (both were missing something).

**Comment by mia-allers-gds** [comment]
Thanks for this @selfthinker. Here are my thoughts:

Header border in relation to the cookie banner border

- I think the header border should match the service nav border, it will look consistent and neat
- I think the cookie banner border should stay as-is (keep the thick border) because it's a separate 'chunk' of the page and in my opinion differentiating it from the header is a good thing

**Comment by querkmachine** [comment]
@mia-allers-gds As an FYI, the way we would normally implement dividing lines for forced colour modes is by adding a transparent border to the component. In this case, that means increasing the height of the header by another pixel.

Do you imagine there'll be any pushback on that or is it okay given it's for accessibility purposes?

**Comment by selfthinker** [comment]
> as it won't be a visible change to the default view I don't see it being an issue.

From what I understand, adding a transparent border is a visible change of that additional space of 1 pixel. Although the border itself won't be visible, its space will be.
But there are ways around that, either by using `outline` instead of `border`, which has the potential to make it look a bit wonky, but most probably not when it's just 1px. Or, what's added to the transparent border could also be removed elsewhere to retain the height. (Although I don't know the code, so it's possible that might make it more complex, or would require lots of re-testing, etc.)

**Comment by mia-allers-gds** [comment]
>Although the border itself won't be visible, its space will be.

As soon I pressed comment I thought about this. It might not be noticeable though, in theory the things likely to appear after the header are phase banners, breadcrumbs and the service nav. Only the service nav has a background colour, so a 1px white border may not be an issue

**Comment by romaricpascal** [comment]
@selfthinker For the footer, we can easily set a `fill="currentColor"` to the crown to become the text colour like the OGL logo. 

For the crest, I had a play with [`mix-blend-mode`](https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode). It does work in High Contrast, however doesn't allow to improve the contrast much. Best I could get was with `plus-lighter`, which is a [fairly recent addition](https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode#browser_compatibility).

<img width="1503" alt="Image" src="https://github.com/user-attachments/assets/2fcc410c-9b6b-42cb-9f69-f1038bac78e3" />

What we could easily do, though, is cancel the `filter` we're using to recolour the SVG, reverting it back to its original light-grey.

<img width="1496" alt="Image" src="https://github.com/user-attachments/assets/76044a4b-b970-44cd-b112-c4697c83f80c" />

Would that be enough of a fix? I'm thinking that may also run into issues for a person that's using forced colours to set a background colour that'd be too close to the one of the original crest (though I think that's an issue we'll have whichever colour we set the crest to, as the user may always pick a colour close to that of the crest).

**Comment by querkmachine** [comment]
@selfthinker @mia-allers-gds Discussed on Slack, but to clarify here too, a 1px transparent border should not create a visible gap between elements. Borders are additive to the component's height[^1], so visually, it'll extend the header's height and background by 1px, not add a 1px gap below it.

[^1]: With the box model we're using.

**Comment by 36degrees** [comment]
> For the crest, I had a play with [`mix-blend-mode`](https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode). It does work in High Contrast, however doesn't allow to improve the contrast much. Best I could get was with `plus-lighter`, which is a [fairly recent addition](https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode#browser_compatibility).

I think I've got a fix for the crest that lets us use `currentColor` for its fill and avoids the `filter` entirely – happy for me to open a PR? It will need a bit of testing but works well with overridden colours in Firefox.

**Comment by selfthinker** [comment]
I've re-tested a [page that has all header-related components in them](https://github.com/alphagov/govuk-frontend/pull/5864).
I haven't found anything new.

The footer images have all been fixed to use the text colour.
And although there is still no border under the header, it's been fixed and just waiting to be merged as part of [adjustments to the header](https://github.com/alphagov/govuk-frontend/pull/5860).

Therefore I can close this now as done.
